test/Quake/basic-sroa.qke

```diff
@@ -0,0 +1,73 @@
+// ========================================================================== //
+// Copyright (c) 2022 - 2025 NVIDIA Corporation & Affiliates.                 //
+// All rights reserved.                                                       //
+//                                                                            //
+// This source code and the accompanying materials are made available under   //
+// the terms of the Apache License 2.0 which accompanies this distribution.   //
+// ========================================================================== //
+
+// RUN: cudaq-opt -cc-sroa %s | FileCheck %s
+// RUN: cudaq-opt -canonicalize -cc-sroa %s | FileCheck %s
+
+func.func private @g(i32, f64)
+
+func.func @f() {
```

Comment:
Is it possible to add a test for a nested array as well?

Reply:
Will do on next PR.